File: aidlserver/src/main/java/com/andrewtse/aidlserver/MyAidlService.java
package com.andrewtse.aidlserver;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.os.RemoteException;
import android.widget.Toast;

/**
 * @author xk
 * @date 2018/12/9
 */
public class MyAidlService extends Service {

    public MyAidlService() {
    }

    @Override
    public IBinder onBind(Intent intent) {
        return myBinder;
    }

    private IBinder myBinder = new IMyServiceInterface.Stub() {
        @Override
        public void basicTypes(int anInt, long aLong, boolean aBoolean, float aFloat, double aDouble, String aString) throws RemoteException {

        }

        @Override
        public void myToast(String content) throws RemoteException {
            Toast.makeText(getApplicationContext(), content, Toast.LENGTH_SHORT).show();
        }
    };
}

File: aidlserver/src/main/java/com/andrewtse/aidlserver/MyAidlService.kt
package com.andrewtse.aidlserver

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.os.RemoteException
import android.widget.Toast

/**
 * @author xk
 * @date 2018/12/9
 */
class MyAidlService : Service() {
    override fun onBind(intent: Intent): IBinder? {
        return myBinder
    }

    private val myBinder: IBinder = object : IMyServiceInterface.Stub() {
        @Throws(RemoteException::class)
        override fun basicTypes(anInt: Int, aLong: Long, aBoolean: Boolean, aFloat: Float, aDouble: Double, aString: String) {
        }

        @Throws(RemoteException::class)
        override fun myToast(content: String) {
            Toast.makeText(applicationContext, content, Toast.LENGTH_SHORT).show()
        }
    }
}